### PR TITLE
MSD - Auto select initial tab without restarting the app

### PR DESCRIPTION
**Pull request by ashiagr**
This PR 
- Cherry picks commits (4859e29c9f871ff0f4812db44a73c4238f9e1fe4...5deee15a7be29519360aae09e681e0c0442ebd3c) from `trunk` that auto-select the default tab without restarting the app on changing `Initial Screen`  from `App Settings` and fix "auto-scroll" to quick start focus point on `Menu` tab for `Stats` and `Pages` items: https://github.com/wordpress-mobile/WordPress-Android/pull/16361#issuecomment-1104744839.
- Adds a precautionary delay before passing navigation actions to nested (view pager) tab fragments to give them time to get created as sometimes it was noticed that `Quick Start Prompt Dialog` was not shown in UI connected test on multiple logins (p1651048630539109/1651048455.086839-slack-C011BKNU1V5).

To test:

#### Test.1
Cherry picked commits (4859e29c9f871ff0f4812db44a73c4238f9e1fe4...5deee15a7be29519360aae09e681e0c0442ebd3c) were tested before merging to `trunk`. Still you can test that changing initial screen from app settings and returning to my site screen auto selects corresponding tab.

#### Test.2
Logout and login multiple times and notice that quick start flow is started every time.

Targets release/19.7
/cc @AliSoftware 

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by AjeshRPai — APPROVED** [review]
I have looked at the code changes and tested it as per the instructions. Everything looks good to me. Great Job @ashiagr. 👍 🙌 .

**Comment by ashiagr** [comment]
👋 @ovitrif 

As discussed here (p1651147755009519/1651048455.086839-slack-C011BKNU1V5), I'll wait to merge to give you a chance to give it a quick look. Note that it needs to be merged to the frozen branch and the sooner we do the better it will be for @AliSoftware to create a new beta.

Thank you for volunteering to lend another pair of eyes. 🙇‍♀️

**Review by ovitrif — APPROVED** [review]
This looks good to me, thank you @ashiagr for your work 🙇, I'll unignore the two login-with-magic-link UI tests afterwards and open a PR, then run the CI step for UI tests a few time to see what we get 🤞 .

I didn't test it but the code looks good. I'll do that async, no need to block the merge 👍

**Comment by ovitrif** [comment]
> I didn't test it but the code looks good. I'll do that async, no need to block the merge 👍

Did that now, It works as expected 🎉 🚀

**Comment by AliSoftware** [comment]
This fix has now been shipped as part of `19.7-rc-4` 🎉

**Pull request merged**
